src/parser/document.ts

```diff
@@ -121,3 +161,26 @@ export interface StringifyOptions {
    */
   inlineDocuments?: ParsedDocument<any, any>[];
 }
+
+/**
+ * The variant of binary search that returns the number of elements in the
+ * array that is strictly less than the target.
+ */
+export function binarySearch(target: number, arr: number[]) {
```

Comment:
If we're going to export this, can it go somewhere else? Seems strange to live in `document.ts`